feat(cloudcontrol): implement rate limit w/ better logging

ekristen opened the pull request:
## Overview

Introducing rate limiting to cloud control api resources. It is a global rate limit for cloud control so all resources leverage it that are cloud control based. This introduces a global rate limit of 25 req/sec to all cloud control describe resource calls.

## Builds

https://github.com/ekristen/aws-nuke/actions/runs/12755642492

## References
- Resolves #508

ekristen commented:
@corybekk please check out the builds from this action https://github.com/ekristen/aws-nuke/actions/runs/12720114622, this introduces a rate limit of 25 request per second, which in testing seems to be just under the allowed threshold. Give it a shot against your entire setup. Hopefully this will resolve the issue for you.

corybekk commented:
> If you are seeing errors, any trace logs on if it's during Describe or Delete? I can lower the rate limit more for testing. I can also add a Delete rate limit as well.

@ekristen using `debug` log level, it shows that the throttling exceptions are occurring during the "list" process.  An Example log:

```
time="2025-01-13T18:58:51Z" level=debug msg="attempting to run lister" owner=us-west-2 resource_type="AWS::ElasticBeanstalk::ApplicationVersion"
time="2025-01-13T18:58:55Z" level=error msg="listing failed:\n    ThrottlingException: Rate exceeded" error="ThrottlingException: Rate exceeded" owner=us-west-2 resource_type="AWS::ElasticBeanstalk::ApplicationVersion"
```

ekristen commented:
@corybekk I've kicked off new builds https://github.com/ekristen/aws-nuke/actions/runs/12755553268

I dropped the rps to 20/sec. I also added a **temporary** env var called `AWS_NUKE_CLOUD_CONTROL_DESCRIBE_RATE_LIMIT` this will allow you to adjust the rate limit further without new builds. If you have time, it would be great to see if 20/s works, if not try 15, then 10 and then maybe in between? 

Some APIs are per minute, so we might need to change the duration to per-minute and try that way. You can drop all the way to 1 or 2 `AWS_NUKE_CLOUD_CONTROL_DESCRIBE_RATE_LIMIT=1` for example, that'd be 60 req/minute.

ekristen commented:
ok, so the only thing else we can try is 59/minute and see if it's per second or per minute, you open to testing one more build?

corybekk commented:
> @ekristen  you open to testing one more build?
Yes 👍

ekristen commented:
@corybekk builds are under way, https://github.com/ekristen/aws-nuke/actions/runs/12757832267 - I should make it so I can build single targets so the builds are quicker, oh well, unfortunately got to wait about 10 minutes. I set it to 55 reqs/minute.

ekristen commented:
@corybekk ok binaries are in https://github.com/ekristen/aws-nuke/actions/runs/12757832267/artifacts/2425403501 -- please let me know the results and I should be able to make one final tweak to the PR and merge it.

Side Note: if we find out there are deletion rate limiting during us, let me know we can add that control in as well.

corybekk commented:
@ekristen that seems to have done the trick. I was able to execute aws-nuke multiple times without running into a throttling exception 🥳

ekristen commented:
Sweet. As soon as the github outage is fixed I'll do a bit of cleanup, and merge it in, you should see a new version later today! Thanks for the testing. Glad we got this in.

ekristen commented:
:tada: This PR is included in version 3.44.0 :tada:

The release is available on [GitHub release](https://github.com/ekristen/aws-nuke/releases/tag/v3.44.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
